src/com/palmergames/bukkit/towny/object/Town.java

```diff
@@ -129,13 +131,32 @@ public double getTaxes() {
 		return taxes == -1 ? TownySettings.getTownDefaultTax() : taxes;
 	}
 
-	public void setMayor(Resident mayor) throws TownyException {
+	/**
+	 * Sets a resident to become mayor. Used only in database loading.
+	 * 
+	 * @param mayor - Town Resident to make into mayor.
+	 * @throws TownyException - When given mayor is not a resident.
+	 */
+	public void forceSetMayor(Resident mayor) throws TownyException {
```

Comment:
I don't really see a necessity in making a separate method just to throw an exception. Maybe just make the `setMayor(...)` method return a boolean indicating success and then handle the result in the loading areas.

Reply:
I find it's a lot easier to have a brute-force method that we know will work, especially for loading things.

Reply:
I mean both methods do the same thing. The only difference is that one throws an exception for one specific check, while the other just black-holes it. I'm just personally against the design practice of creating one slightly-different method that's only supposed to be in like 1-2 specific places.

Reply:
Looked over this again. The force one is for loading only, has an exception that is caught if Towny needs to rethink who should be mayor/if a mayor is possible at all.

The second one doesn't throw an exception on purpose, one part of the update was making it so that during runtime the setMayor() will never throw an exception, making for less headaches.